### PR TITLE
dev-db/redis: rework src_test

### PR DESCRIPTION
**Pull request by arkamar**
I think `die` should be used with all those scripts or is there a reason why it shouldn't?

### CODE REVIEW

**Comment by arkamar** [comment]
I will append https://bugs.gentoo.org/649868 fix to this PR as well. I'm working on it.

**Comment by arkamar** [comment]
It took slightly longer than I thought but I believe it was for good thing. All tests seem to work now with usual default `FEATURES`.

File: dev-db/redis/redis-6.0.16.ebuild [hunk 1]
```diff
@@ -133,18 +130,31 @@ src_compile() {
...
 	if use ssl; then
-		./utils/gen-test-certs.sh
-		./runtest --tls
-	else
-		./runtest
+		./utils/gen-test-certs.sh || die "Test certs generetaion failed"
+		runtestargs+=( --tls )
 	fi
+
+	echo "./runtest ${runtestargs[@]}"
```
Comment (thesamesam) [inline dev-db/redis/redis-6.0.16.ebuild:156]: edo! :)

File: dev-db/redis/redis-7.0.1.ebuild [hunk 2]
```diff
@@ -55,9 +55,6 @@ PATCHES=(
 src_prepare() {
 	default
 
-	# unstable on jemalloc
```
Comment (thesamesam) [inline dev-db/redis/redis-7.0.1.ebuild:58]: We've lost this, is that ok?

File: dev-db/redis/redis-6.0.16.ebuild (hunk 1 above)
Comment (arkamar) [reply dev-db/redis/redis-6.0.16.ebuild]: oh, nice, I have changed that.

File: dev-db/redis/redis-7.0.1.ebuild (hunk 2 above)
Comment (arkamar) [reply dev-db/redis/redis-7.0.1.ebuild:58]: Yes, it is ok. Sorry, I was too tired and lazy to write some summary of those commits yesterday, but it is actually explained in commit messages. This issue seems to affect `6.0` only, see https://github.com/gentoo/gentoo/pull/25831/commits/fefe8b0398f983501e754ddd1444811d2ac315fb. It was fixed in `6.2*` and it does not affect `7.0*` either, at least those tests work for me, see https://github.com/gentoo/gentoo/pull/25831/commits/b325b5d9f167f9ed34de76b87afa48ec73ae3000.

**Comment by arkamar** [comment]
Summary of last FP:
- use of `edo` for all local test script calls
- Use `MAKEOPTS` jobs to limit number of test clients, this change is in two commits, first one changes `5.0.14` only because it was necessary to introduce `src_test` function and the second addapts the rest of ebuilds where `./runtest` script is already used.
- defrag tests are restored in another commit in version `6.2*` and `7.0*`
- only defrag tests are skipped in `6.0` in next commit
- the last commit resolves problem with `oom-score-adj` related tests when `usersanbox` or `-userpriv` `FEATURES` are set.

File: dev-db/redis/redis-7.0.1.ebuild (hunk 2 above)
Comment (thesamesam) [reply dev-db/redis/redis-7.0.1.ebuild:58]: It's my bad for not reading the commit messages! Great work here.

**Comment by thesamesam** [comment]
Big thank you!

**Comment by thesamesam** [comment]
Also, if you're interested, you're free to join me in maintaining Redis.

**Pull request closed without merging**